tests/test_spectral_nomination.py

```diff
@@ -13,156 +19,94 @@
+            self._nominate(np.zeros((5, 5, 5), dtype=np.int), np.zeros(3, dtype=np.int))
+        with self.assertRaises(TypeError):
+            self._nominate([[0] * 10] * 10, np.zeros(3, dtype=np.int))
+        # embedding should have fewer cols than rows.
+        svn = SpectralVertexNomination(input_graph=False)
+        with self.assertRaises(IndexError):
+            self._nominate(
+                np.zeros((10, 20), dtype=np.int),
+                np.zeros(3, dtype=np.int),
+                nominator=svn,
+            )
+        # adj matrix should be square
+        with self.assertRaises(IndexError):
+            self._nominate(np.zeros((3, 4), dtype=np.int), np.zeros(3, dtype=np.int))
 
     def _test_k(self):
-        _test_k_value()
-        _test_k_type()
+        # k should be > 0
+        with self.assertRaises(ValueError):
+            self._nominate(adj, np.zeros(3, dtype=np.int), k=0)
+        # k of wrong type
+        with self.assertRaises(TypeError):
+            self._nominate(adj, np.zeros(3, dtype=np.int), k="hello world")
 
     def test_constructor_inputs(self):
-        _test_embedder_value()
+        with self.assertRaises(ValueError):
+            svn = SpectralVertexNomination(embedder="hi")
+            self._nominate(adj, np.zeros(3, dtype=int), nominator=svn)
 
     def test_constructor_inputs1(self):
-        _test_embedder_type()
+        # embedder must be BaseSpectralEmbed or str
+        with self.assertRaises(TypeError):
+            svn = SpectralVertexNomination(embedder=45)
 
     def test_constructor_inputs2(self):
-        _test_input_graph_bool_type()
+        # input graph param has wrong type
+        with self.assertRaises(TypeError):
+            svn = SpectralVertexNomination(input_graph=4)
 
-    @pytest.mark.parametrize(
-        "nominator",
-        [
+    def test_basic_unattributed(self):
+        """
+        Runs two attributed seeds and two unattributed seeds with each nominator.
+        Ensures all options work. Should be fast. Nested parametrization tests all
+        combinations of listed parameters.
+        """
+        nominators = [
             SpectralVertexNomination(embedder="ASE"),
             SpectralVertexNomination(embedder="LSE"),
             SpectralVertexNomination(embedder=embeder),
-        ],
-    )
-    @pytest.mark.parametrize(
-        "seed",
-        [
+        ]
+        seeds = [
             np.array([8]),
             np.array([2, 6, 9, 15, 25]),
             np.arange(n_verts - 1, dtype=np.int),
-        ],
-    )
-    def test_basic_unattributed(self, nominator, seed):
-        """
-        Runs two attributed seeds and two unattributed seeds with each nominator.
-        Ensures all options work. Should be fast. Nested parametrization tests all
-        combinations of listed parameters.
-        """
-        _nominate(adj, seed, nominator)
+        ]
+        for nominator, seed in itertools.product(nominators, seeds):
```

Comment:
How to hide a nested for loop in one easy step! ;D

Reply:
> How to hide a nested for loop in one easy step! ;D

I have used this trick to make code ugly before more than once, not gonna lie...